### ISSUE TITLE
Add `py.typed` file to `gradio` package

### ISSUE THREAD

**Issue by aleneum** [gradio-app/gradio#6396]
- [x] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
Whenever I use `mypy` I will get the following warning:

 ```
Skipping analyzing "gradio": module is installed, but missing library stubs or py.typed marker
```

As far as I know, there are no Gradio stubs and Gradio does not contain a `py.typed` file as well.
A common workaround for untyped packages is to tell `mypy` to not follow such imports:

```
[mypy-gradio.*]
ignore_missing_imports = True
```

However, when `disallow_any_unimported` is enabled, `mypy` cannot type check functions with Gradio parameters:

```
Argument 2 to "upload" becomes "Any | None" due to an unfollowed import  [no-any-unimported]
```

My current solution is to allow any imports whenever Gradio is used. However, as most `gradio` imports seem to be typed, why not just add a `py.typed` file to the package to be [PEP 561](https://peps.python.org/pep-0561/#packaging-type-information) compliant? This would make it easier to spot typing issues when working with Gradio.

**Describe the solution you'd like**  
Add an empty file `py.typed` to the package (`<project>/gradio`) and include it via:

```toml
[tool.setuptools.package-data]
gradio = ["py.typed"]
```

I am happy to create a PR for this (hopefully trivial) change when you consider this useful.

**Comment by abidlabs** [comment]
Thanks @aleneum this seems like a helpful addition -- please go ahead and create the PR!

**Comment by aleneum** [comment]
Hello @abidlabs,

it is (of course) not as straight forward as I have hoped for. 

Problem 1: Adding a `py.typed` file to gradio will break `pip install .` (from source) since necessary `pyi`-stubs are generated only after `gradio` has been imported in `format_backend.sh`.

Solution: I could create/touch an empty file the moment a `pyi` file in generated or inside `format_backend.sh`. 

Problem 2: In [templates.py](https://github.com/gradio-app/gradio/blob/0bafdcb550feefceb984e910c01c15d9d398ee1b/gradio/templates.py), there are multiple classes commented out. The problem is that some/all of the contain an attribute named 'type'. `mypy` (and maybe other type checkers) use the annotation '#  type:' for inline configuration. Consequently, a commented out type annotation such as `#        type: int` will cause syntax issues [ref](https://github.com/python/mypy/issues/14945)

Solution: I could comment out the relevant classes in templates.py twice and add a remark concerning this issue.

**Comment by JackismyShephard** [comment]
Not sure this is the right place to comment, but I am still having typing issues, although with pyright, not mypy. I am getting a `Stub file not found for "gradio"` warning, when importing gradio.

**Comment by abidlabs** [comment]
Hmm that's strange. Can you create a separate issue with step-by-step details on how to reproduce?

**Comment by JackismyShephard** [comment]
@abidlabs I will try
